README.md

```diff
@@ -202,6 +202,23 @@ pytest  # run test in current environment
 If you chose to create docs with your plugin read the corresponding docs
 for [Sphinx] or [MkDocs]
 
+### Pre-commit
+
+This template includes a default yaml configuration for [pre-commit](https://pre-commit.com/).
+Among other things, it includes checks for best practices in napari plugins.
+You may edit the config at `.pre-commit-config.yaml`
+
+To use it run:
+
+```bash
+pip install pre-commit
+pre-commit install
+```
+
+You can also have these checks run automatically for you when you push to github
+by installing [pre-commit ci](https://pre-commit.ci/) on your repository.
+
+
```

Comment:
I think it's worth having a short sentence or two here about what to do if the pre-commit check fails. It can be very disconcerting for new users when they *think* they are making a commit and pushing, but nothing is actually being recorded (because pre-commit is failing).

Nothing fancy, but I think we should say that if pre-commit fails any of the checks that's ok. Most of the time it will automatically modify your code file to fix the problem, so all you need to do is `git add name/of/files` and then try committing again.

Although now I say that, do all the pre-commit checks modify the files, or are there some where you have to do it yourself? (I've seen that happen before too, mostly with things like flake8 finding unused imports, not sure if that's to be expected here as well).

Reply:
> do all the pre-commit checks modify the files, or are there some where you have to do it yourself?

this depends on the hook.  These hooks are generally just running some command line thing underneath the hood, so the question must be asked for each of them. flake8 and mypy (and `napari-plugin-checks`) are _checkers_, they simply tell you when something is wrong.  setup-cfg-fmt, autoflake, isort, black, and pyupgrade are all _formatters_.  These are tools that modify your code.